app/src/main/java/com/example/demoquizwithandroid/QuizActivity.kt

```diff
@@ -13,63 +12,62 @@ import android.widget.RadioGroup
-                    findViewById<TextView>(R.id.questionNumber).text = "Question ${(index+1)}/${questions.length()}"
-                    currentQuestion = questions[index] as JSONObject
+        viewModel.questions.observe(this) { questions ->
+            if (questions != null) {
+                viewModel.selectedQuestion.observe(this) { index ->
+                    findViewById<TextView>(R.id.questionNumber).text =
+                        "Question ${(index + 1)}/${questions.size}"
+                    currentQuestion = questions[index]
 
-                    val options = currentQuestion!!["options"] as JSONArray
-                    findViewById<TextView>(R.id.question).text = currentQuestion!!["question"].toString()
-                    findViewById<RadioButton>(R.id.option1).text = options[0].toString()
-                    findViewById<RadioButton>(R.id.option2).text = options[1].toString()
-                        val option3= findViewById<RadioButton>(R.id.option3)
-                        val option4=findViewById<RadioButton>(R.id.option4)
-                    if(options.length() > 2) {
-                            option4.text = options[3].toString()
-                            option3.text = options[2].toString()
+                    val options = currentQuestion!!.options
+                    findViewById<TextView>(R.id.question).text =
+                        currentQuestion!!.question
+                    findViewById<RadioButton>(R.id.option1).text = options[0]
+                    findViewById<RadioButton>(R.id.option2).text = options[1]
+                    val option3 = findViewById<RadioButton>(R.id.option3)
+                    val option4 = findViewById<RadioButton>(R.id.option4)
+                    if (options.size > 2) {
+                        option4.text = options[3]
+                        option3.text = options[2]
                         option3.visibility = View.VISIBLE
                         option4.visibility = View.VISIBLE
                     } else {
                         option3.visibility = View.GONE
                         option4.visibility = View.GONE
 
                     }
-                })
+                }
             }
-        })
+        }
         findViewById<Button>(R.id.btnNext).setOnClickListener {
             _ -> onNextClicked()
         }
     }
 
     private fun onNextClicked() {
         if(currentQuestion == null) return
-        val rbGroup : RadioGroup = findViewById<RadioGroup>(R.id.options)
+        val rbGroup = findViewById<RadioGroup>(R.id.options)
         val rbSelected : RadioButton = findViewById(rbGroup.checkedRadioButtonId) ?: return
-        val ans: String = rbSelected.text.toString();
-        if (ans == currentQuestion!!["answer"].toString()) {
+        val ans: String = rbSelected.text.toString()
+        if (ans == currentQuestion!!.answer) {
             viewModel.correct.postValue(viewModel.correct.value?.inc())
```

Comment:
I think this is causing an error because of dispatching to main thread.

I think this should be
```diff
-viewModel.correct.postValue(viewModel.correct.value?.inc())
+viewModel.correct.value = viewModel.correct.value?.inc()
```

Reply:
You are right, this was causing the problem, fixed.